uPortal-groups/uPortal-groups-pags/src/main/java/org/apereo/portal/groups/pags/testers/InvertedRegexTester.java

```diff
@@ -49,27 +49,15 @@
  * </table>
  * </code>
  */
-public class InvertedRegexTester extends AbstractStringTester {
-
-    protected Pattern pattern;
+public class InvertedRegexTester extends RegexTester {
 
     /** @since 4.3 */
     public InvertedRegexTester(IPersonAttributesGroupTestDefinition definition) {
         super(definition);
-        this.pattern = Pattern.compile(definition.getTestValue());
-    }
-
-    /**
-     * Sets the pattern string to use for the regex test.
-     *
-     * @param patternString regex pattern string
-     */
-    protected void setPattern(String patternString) {
-        pattern = Pattern.compile(patternString);
     }
 
     @Override
-    public boolean test(String att) {
-        return !pattern.matcher(att).matches();
+    public boolean test(IPerson person) {
```

Comment:
I'm curious why this tests a person rather than an attribute

Reply:
It's a change that Drew made when migrate PAGS from xml to JPA, it's better as you can access to all IPerson attributes in an abstract way (It implements the [IPersonTester interface](https://github.com/Jasig/uPortal/blob/master/uPortal-groups/uPortal-groups-pags/src/main/java/org/apereo/portal/groups/pags/IPersonTester.java)).

The oldest method boolean test(String att) isn't anymore used expect internally in some cases.

Reply:
thanks for the clarity